prototype/core/geo-json/src/main/java/org/eclipse/sensinact/gateway/geojson/GeoJsonObject.java

```diff
@@ -61,4 +63,44 @@ protected GeoJsonObject(GeoJsonType type) {
     public Map<String, Object> foreignMembers = new HashMap<>();
 
     public List<Double> bbox;
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(type, bbox);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof GeoJsonObject) {
+            final GeoJsonObject other = (GeoJsonObject) obj;
+            return type == other.type && Objects.equals(bbox, other.bbox)
+                    && Objects.equals(foreignMembers, other.foreignMembers);
+        }
+
+        return false;
+    }
+
+    /**
+     * Returns the object description
+     */
+    protected abstract String getObjectDescription();
+
+    @Override
+    public String toString() {
+        final List<String> parts = new ArrayList<>();
```

Comment:
If we start with a StringBuilder we can pass it to the subclass method to populate.

Also, shouldn't we start by putting the `type` into the list/StringBuilder so we know what type of Object it is?